grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy

```diff
@@ -55,48 +48,61 @@ class SubmissionController extends AuthenticatedController {
             }
         }
         SubmissionType defaultType = SubmissionType.getForLabel(params.get("type").toString()) ?: SubmissionType.Amendment.label
-        render(view: '/submission/submission',
-                model: [issue:                      issue,
-                        submission:                 submission,
-                        docTypes:                   PROJECT_DOC_TYPES,
-                        submissionTypes:            submissionTypes,
-                        submissionNumberMaximums:   submissionNumberMaximums,
-                        defaultType:                defaultType
-                ])
+        render( [issue:                      issue,
+                 typeLabel:                  issue.typeLabel,
+                 submission:                 submission,
+                 documents:                  submission?.documents,
+                 docTypes:                   PROJECT_DOC_TYPES,
+                 submissionTypes:            submissionTypes,
+                 submissionNumberMaximums:   submissionNumberMaximums,
+                 defaultType:                defaultType
+        ] as JSON)
     }
 
     def save() {
-        Issue issue = queryService.findByKey(params.projectKey)
-        Submission submission
-        if (params?.submissionId) {
-            submission = Submission.findById(params?.submissionId)
-            submission.number = Integer.parseInt(params?.submissionNumber)
-            submission.type = params?.submissionType
-            submission.comments = params?.comments
-            if (!submission.save(flush: true)) {
-                flash.message = submission.errors.allErrors.collect { it }.join("<br/>")
+        JsonParser parser = new JsonParser()
+        JsonArray dataSubmission = parser.parse(request.parameterMap["submission"].toString())
+        List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
```

Comment:
Here and above, you might want to use the null-safe operator (`?`) to traverse some of these objects that might not really be part of the request.